robosherlock/src/io/src/MongoDBBridge.cpp

```diff
@@ -81,79 +83,113 @@ void MongoDBBridge::readConfig(const boost::property_tree::ptree &pt)
 
 bool MongoDBBridge::setData(uima::CAS &tcas, uint64_t timestamp)
 {
+  ros::NodeHandle n_("~");
+  double_t time_stamp; // Variable which store the value of queued time stamp 
+  
+  n_.getParam("ts",time_stamp); // getting the value of "ts" from the command arg and store in time_stamp.
+  n_.deleteParam("ts"); // deleting the "ts" variable.
+  
   MEASURE_TIME;
   const bool isNextFrame = timestamp == std::numeric_limits<uint64_t>::max();
-
-  if(actualFrame >= frames.size()) {
-    if(continual) {
-      storage.getScenes(frames);
-      if(actualFrame >= frames.size()) {
-        return false;
-      }
-    }
-    else if(loop) {
-      actualFrame = 0;
-    }
-    else {
-      outInfo("last frame. shuting down.");
-      cv::waitKey();
-      ros::shutdown();
-      return false;
-    }
-  }
-  if(isNextFrame) {
-    outInfo("default behaviour");
-    timestamp = frames[actualFrame];
-    outInfo("setting data from frame: " << actualFrame << " (" << timestamp << ")");
+  
+  auto it= std::find(frames.begin(), frames.end(), time_stamp); // finding if the queued timestamp is in the frames vector
+  
+  if(it== frames.end()){
+      outWarn("Timestamp Queried is not in the frame or Timestamp Query not enabled");
```

Comment:
This is triggered on every iteration when the ts parameter is not set by the user.
Could you please change the method so that it checks if the timestamp parameter is set and only then do the required steps?